src/MongrationDotNet/MigrationRunner.cs

```diff
@@ -29,33 +31,54 @@ public async Task Migrate()
-                        migrationCollection.Type, migrationCollection.Version.ToString(),
-                        migrationCollection.Description);
-                    continue;
-                }
+            if (latestAppliedMigration != null)
+            {
+                logger?.LogInformation(LoggingEvents.MigrationSkipped,
+                    "Migration has already been applied or in progress. Skipped migration for type: {type}, version: {version} and description: {description} ",
+                    migrationCollection.Type, migrationCollection.Version.ToString(),
+                    migrationCollection.Description);
+                continue;
+            }
 
-                await SetMigrationInProgress(migrationCollection);
-                migrationCollection.Prepare();
-                await migrationCollection.ExecuteAsync(database, logger);
-                await SetMigrationAsCompleted(migrationCollection);
-                logger?.LogInformation(LoggingEvents.MigrationCompleted,
-                    "Migration completed type: {type}, version: {version} and description: {description} ",
-                    migrationCollection.Type, migrationCollection.Version.ToString(), migrationCollection.Description);
+            await SetMigrationInProgress(migrationCollection);
+            migrationCollection.Prepare();
+            await migrationCollection.ExecuteAsync(database, logger);
+            await SetMigrationAsCompleted(migrationCollection);
+            logger?.LogInformation(LoggingEvents.MigrationCompleted,
+                "Migration completed type: {type}, version: {version} and description: {description} ",
+                migrationCollection.Type, migrationCollection.Version.ToString(), migrationCollection.Description);
             }
         }
 
         private async Task SetMigrationInProgress(IMigration migrationCollection)
         {
-            await migrationDetailsCollection.InsertOneAsync(migrationCollection.MigrationDetails);
+            try
+            {
+                await migrationDetailsCollection.InsertOneAsync(migrationCollection.MigrationDetails);
+            }
+            catch (MongoWriteException)
+            {
+               await PollForMigrationStatus(migrationCollection);
+            }
+        }
+
+        private async Task PollForMigrationStatus(IMigration migrationCollection)
+        {
+            MigrationDetails migrationDetails;
+            var counter = 0;
+            do
+            {
+                counter ++;
+                await Task.Delay(5000);
+                migrationDetails = await migrationDetailsCollection
+                    .Find(x => x.Version == migrationCollection.Version && x.Status == "Completed")
+                    .FirstOrDefaultAsync();
+            } while (migrationDetails == null || counter<24);
```

Comment:
Use the Options pattern so that library users can configure that polling interval and the timeout duration before a migration is considered as failed.

Add some debug logging for the polling.

If polling for migration status times out (counter = 24 here) then what should happen? Throw exception? stop and skip all other migrations? Apply the next migration? Retry the timed out migration? Will let you decide which strategy!